electroncash/rpa/addr.py

```diff
@@ -0,0 +1,180 @@
+# Copyright (c) 2017 Pieter Wuille
+# Copyright (c) 2017 Shammah Chancellor, Neil Booth
+# Copyright (c) 2020 Jonald Fyookball
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+"""
+Implements a custom cashaddr-style encoding for a Reusable Payment Address (RPA)
```

Comment:
I recall discussing this with @cculianu a while back.  I think we specifically decided to NOT consolidate it and have it be its own section of code because it didn't conform to the cashaddr spec.  I don't remember the exact reason.  Maybe some of the cashaddr error checks were at issue, as you alluded to.  Could have been because the hash size was too big.  In any case, I'm ok with having it be its own section for now.

Reply:
Honestly @jonas-lundqvist is right.  The copy-pasta here is just lazy and terrible.

Don't invoke my name here.  I never agreed to this.  I just didn't stop you from doing it for want of defending the rest of the codebase from modification (and potential breakage).